manta-js/package/sdk.ts

```diff
@@ -0,0 +1,604 @@
+            this.walletIsBusy = false;
+            console.error("Failed to build transaction.",e);
+        }
+    }
+
+    /// Executes a "Private Transfer" transaction for any fungible token.
+    async privateTransferSend(assetId: BN, amount: BN, toPrivateAddress: Address, polkadotSigner:Signer, polkadotAddress:Address): Promise<void> {
+        const signed = await this.privateTransferBuild(assetId,amount,toPrivateAddress,polkadotSigner,polkadotAddress);
+        await this.sendTransaction(polkadotAddress,signed);
+        this.conditionalLog("Private Transfer transaction finished.");
+    }
+
+    /// Builds a "Private Transfer" transaction for any fungible token.
+    /// Note: This transaction is not published to the ledger.
+    async privateTransferBuild(assetId: BN, amount: BN, toPrivateAddress: Address, polkadotSigner:Signer, polkadotAddress:Address): Promise<SignedTransaction> {
+        try {
+            await this.waitForWallet();
+            this.walletIsBusy = true;
+            await this.setPolkadotSigner(polkadotSigner, polkadotAddress);
+            const transaction = await this.privateFungibleTransfer(assetId, amount, toPrivateAddress);
+            const signResult = await this.signTransaction(transaction.assetMetadataJson, transaction.transaction, this.network);
+            this.walletIsBusy = false;
+            return signResult;
+        } catch (e) {
+            this.walletIsBusy = false;
+            console.error("Failed to build transaction.",e);
+        }
+    }
+
+    /// Executes a "To Public" transaction for any fungible token.
+    async toPublicSend(assetId: BN, amount: BN, polkadotSigner:Signer, polkadotAddress:Address): Promise<void> {
+        const signed = await this.toPublicBuild(assetId,amount,polkadotSigner, polkadotAddress);
+        await this.sendTransaction(polkadotAddress,signed);
+        this.conditionalLog("To Public transaction finished.");
+    }
+
+    /// Builds and signs a "To Public" transaction for any fungible token.
+    /// Note: This transaction is not published to the ledger.
+    async toPublicBuild(assetId: BN, amount: BN, polkadotSigner:Signer, polkadotAddress:Address): Promise<SignedTransaction> {
+        try {
+            await this.waitForWallet();
+            this.walletIsBusy = true;
+            await this.setPolkadotSigner(polkadotSigner, polkadotAddress);
+            const transaction = await this.toPublicFungible(assetId, amount);
+            const signResult = await this.signTransaction(transaction.assetMetadataJson, transaction.transaction, this.network);
+            this.walletIsBusy = false;
+            return signResult;
+        } catch (e) {
+            this.walletIsBusy = false;
+            console.error("Failed to build transaction.",e);
+        }
+    }
+
+    /// Executes a public transfer.
+    async publicTransfer(assetId: BN, amount: BN, destinationAddress: Address, senderAddress:Address, polkadotSigner:Signer): Promise<void> {
```

Comment:
We should also move this function out of this class/file and into q new file called `utils.ts` since it is not called from within `MantaPrivateWallet`, and not really private wallet functionality.